scripts/compare_benchmarks.py

```diff
@@ -84,74 +144,194 @@ def calculate_and_format_p_value(old, new):
+        new_avg_unsuccessful_iteration = np.mean(new_unsuccessful_durations)
+        if len(old_unsuccessful_durations) > 0 and len(new_unsuccessful_durations) > 0:
+            diff_throughput_unsuccessful = float(new_unsuccessful_per_second / old_unsuccessful_per_second)
+            diff_duration_unsuccessful = new_avg_unsuccessful_iteration / old_avg_unsuccessful_iteration
         else:
-            diff_unsuccessful = float("nan")
+            diff_throughput_unsuccessful = float("nan")
+            diff_duration_unsuccessful = float("nan")
 
         unsuccessful_info = [
-            "  unsucc.:",
-            " " + str(old_unsuccessful_per_second),
-            " " + str(len(old["unsuccessful_runs"])),
-            " " + str(new_unsuccessful_per_second),
-            " " + str(len(new["unsuccessful_runs"])),
-            " " + format_diff(diff_unsuccessful, True),
+            "   unsucc.:",
+            f"{(old_avg_unsuccessful_iteration / 1e6):>7.1f}"
+            if not math.isnan(old_avg_unsuccessful_iteration)
+            else "nan",
+            f"{(new_avg_unsuccessful_iteration / 1e6):>7.1f}"
+            if not math.isnan(new_avg_unsuccessful_iteration)
+            else "nan",
+            format_diff(diff_duration_unsuccessful) + " " if not math.isnan(diff_duration_unsuccessful) else " ",
+            f"{old_unsuccessful_per_second:>.2f}",
+            f"{new_unsuccessful_per_second:>.2f}",
+            format_diff(diff_throughput_unsuccessful) + " " if not math.isnan(diff_throughput_unsuccessful) else " ",
         ]
 
         unsuccessful_info_colored = [colored(text, attrs=["dark"]) for text in unsuccessful_info]
         table_data.append(unsuccessful_info_colored)
 
-table_data.append(["geometric mean", "", "", "", "", format_diff(geometric_mean(diffs)), ""])
+table_data.append(
+    [
+        "Sum",
+        f"{(total_runtime_old / 1e6):>7.1f}",
+        f"{(total_runtime_new / 1e6):>7.1f}",
+        color_diff(total_runtime_new / total_runtime_old, True) + " ",
+    ]
+)
+table_data.append(["Geomean", "", "", "", "", "", color_diff(geometric_mean(diffs_throughput)) + " "])
 
 table = AsciiTable(table_data)
-table.justify_columns[6] = "right"
+for column_index in range(1, len(table_data[0])):  # all columns justified to right, except for item name
+    table.justify_columns[column_index] = "right"
+
+result = str(table.table)
+
+new_result = ""
+lines = result.splitlines()
+
+
+# Double the vertical line between the item names and the two major measurements.
+lines = double_vertical_separators(lines, [1, 4])
+
+# As the used terminaltables module does not support cells that span multiple columns, we do that manually for latency
+# and throughput in the header. We used two space holders that are narrow enough to not grow the column any wider than
+# necessary for the actual values. As the new header spans two columns, we use the full descriptions.
```

Comment:
"we use the full descriptions"?

Reply:
Changed last sentence to "After manually changing the column title to span two column, we replace the place
# holder the actual full descriptions.".

Reply:
replace with?